eng/pipelines/swagger-api-doc-preview.yml

```
@@ -0,0 +1,203 @@
+trigger: none
+pr:
+  paths:
+    include:
+      - specification/**
+
+jobs:
+  - job: SwaggerApiDocPreview
+
+    pool:
+      name: $(LINUXPOOL)
+      vmImage: $(LINUXVMIMAGE)
+      os: linux
+
+    variables:
+      - template: /eng/pipelines/templates/variables/globals.yml
+      - template: /eng/pipelines/templates/variables/image.yml
+      
+      - name: BranchName
+        value: preview/$(Build.Repository.Name)/pr/$(System.PullRequest.PullRequestNumber)/build/$(Build.BuildId)/attempt/$(System.JobAttempt)
+      
+      - name: CurrentBuildUrl
+        value: $(System.CollectionUri)$(System.TeamProject)/_build/results?buildId=$(Build.BuildId)
+
+      # TODO: Remove [TEST-IGNORE] when this is ready to go live
+      - name: StatusName
+        value: '[TEST-IGNORE] Swagger ApiDocPreview'
+
+    steps:
+      - template: /eng/pipelines/templates/steps/set-pr-check.yml
+        parameters:
+          State: pending
+          TargetUrl: $(CurrentBuildUrl)
+          Description: 'Starting docs build'
+          Context: $(StatusName)
+
+      - checkout: self
+        # Fetch depth required to get list of changed files
+        fetchDepth: 2
+
+      - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+        parameters:
+          SkipCheckoutNone: true
+          TokenToUseForAuth: $(azuresdk-github-pat)
+          # Path does not need to be set because sparse-checkout.yml already
+          # checks out files in the repo root
+          Repositories:
+            - Name: MicrosoftDocs/AzureRestPreview
+              # TODO: use main branch when this is ready to go live
+              Commitish: refs/heads/doc-preview-migration-test
+              WorkingDirectory: AzureRestPreview
+
+      - template: /eng/pipelines/templates/steps/npm-install.yml
```

```suggestion
      - template: /eng/pipelines/templates/steps/npm-install.yml
```

What JS deps do you actually need?  Best case is none, next would be just .github (or .github/shared), most expensive is repo root.